conans/test/functional/toolchains/cmake/test_file_api.py

```diff
@@ -0,0 +1,137 @@
+                                   public_header="decoder.h")
+    encoder_cmake = gen_cmakelists(libname="encoder", libsources=["encoder.cpp"], install=True,
+                                   public_header="encoder.h")
+    transcoder_cmake = gen_cmakelists(libname="transcoder", libsources=["transcoder.cpp"], install=True,
+                                      public_header="transcoder.h", deps=["decoder", "encoder"])
+    common_cmake = textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8)
+        project(triunfo)
+        add_subdirectory(decoder)
+        add_subdirectory(encoder)
+        add_subdirectory(transcoder)
+    """)
+
+    client.save({"conanfile.py": conanfile,
+                 os.path.join("decoder", "decoder.cpp"): decoder_cpp,
+                 os.path.join("encoder", "encoder.cpp"): encoder_cpp,
+                 os.path.join("transcoder", "transcoder.cpp"): transcoder_cpp,
+                 os.path.join("decoder", "decoder.h"): decoder_h,
+                 os.path.join("encoder", "encoder.h"): encoder_h,
+                 os.path.join("transcoder", "transcoder.h"): transcoder_h,
+                 os.path.join("decoder", "CMakeLists.txt"): decoder_cmake,
+                 os.path.join("encoder", "CMakeLists.txt"): encoder_cmake,
+                 os.path.join("transcoder", "CMakeLists.txt"): transcoder_cmake,
+                 "CMakeLists.txt": common_cmake,
+                 })
+    client.run("create .")
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        from conan.tools.cmake import CMake
+
+        class Elogio(ConanFile):
+            name = "elogio"
+            version = "1.0"
+            requires = "triunfo/1.0"
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = "*CMakeLists.txt", "*.cpp", "*.h"
+            generators = "CMakeDeps", "CMakeToolchain"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+    """)
+
+    use_decoder_cpp = gen_function_cpp(name="main", includes=["decoder"], calls=["decoder"])
+    use_encoder_cpp = gen_function_cpp(name="main", includes=["encoder"], calls=["encoder"])
+    use_transcoder_cpp = gen_function_cpp(name="main", includes=["transcoder"], calls=["transcoder"])
+    use_decoder_cmake = gen_cmakelists(appname="use_decoder", appsources=["use_decoder.cpp"],
+                                       find_package={"triunfo": "decoder"})
+    use_encoder_cmake = gen_cmakelists(appname="use_encoder", appsources=["use_encoder.cpp"],
+                                       find_package={"triunfo": "encoder"})
+    use_transcoder_cmake = gen_cmakelists(appname="use_transcoder", appsources=["use_transcoder.cpp"],
+                                          find_package={"triunfo": "transcoder"})
+    common_cmake = textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8)
```

Comment:
why? I prefer not to set unnecessarily high requirements, but instead ensure maximum compatibility

Reply:
Because policies change based on this version, and something might not work the same with an older version, it is not the first time something in Conan didn't work as expected because of this version.

Reply:
they do change, but most of conan tests use `2.8` or `2.8.12` right now, so I am not sure, what exactly are we trying to catch here.
cmake files here are pretty simple and don't use any sophisticated features that would require newer release.

Reply:
Many of the new tests of the new integration are already using ``cmake_minimum_required(VERSION 3.15)``, because the behavior when using the generated xxx-config.cmake or conan_toolchain.cmake, was not correct. Maybe in this test not an issue yet, I am fine to leave it, but it might fail later down the road when something change.